retrievalmarket/impl/askstore/askstore_impl.go

```diff
@@ -79,7 +99,40 @@ func (s *AskStoreImpl) GetAsk() *retrievalmarket.Ask {
 		return nil
 	}
 	ask := *s.ask
-	return &ask
+	return ask.Ask
+}
+
+// GetSignedAsk returns the current retrieval ask, signed with the provider's
+// key, or nil if there is no ask set.
+func (s *AskStoreImpl) GetSignedAsk() (*retrievalmarket.SignedRetrievalAsk, error) {
```

Comment:
Why do we need to sign asks ? Also, as commented elsewhere, just returning the ask here without using the pricing function wont be of much use.